src/Compiler/Utilities/ParallelProcessing.fs

```diff
@@ -0,0 +1,53 @@
+﻿module internal FSharp.Compiler.Service.Utilities.ParallelProcessing
+
+#nowarn "1182"
+
+open System
+open System.Collections.Concurrent
+open System.Threading
+open FSharp.Compiler.Diagnostics
+open Internal.Utilities.Library.Extras
+
+/// <summary>
+/// Process items in parallel, allow more work to be scheduled as a result of finished work,
+/// limit parallelisation to 'parallelism' threads.
+/// </summary>
+/// <remarks>
+/// Could be replace with MailboxProcessor+Tasks/Asyncs instead of BlockingCollection + Threads.
+/// See http://www.fssnip.net/nX/title/Limit-degree-of-parallelism-using-an-agent for an example.
+/// </remarks>
+let processInParallel
+    (opName: string)
+    (firstItems: 'Item[])
+    (work: 'Item -> 'Item[])
```

Comment:
I am thinking about appropriate naming here.

The combination of firstItems + work forms a task-graph right ? So 'processGraphInParallel' might be more explicit.

Also is shouldStop supposed to be different from running out of work (i.e. all items finisihed and the last ones returned empty work arrays)

Reply:
This is no longer used, I will remove it.